ganttproject/src/net/sourceforge/ganttproject/action/resource/AssignmentToggleAction.java

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019 BarD Software s.r.o [Juanan Pereira]
+
+This file is part of GanttProject, an opensource project management tool.
+
+GanttProject is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+GanttProject is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GanttProject.  If not, see <http://www.gnu.org/licenses/>.
+*/
+package net.sourceforge.ganttproject.action.resource;
+
+import net.sourceforge.ganttproject.action.GPAction;
+import net.sourceforge.ganttproject.gui.UIFacade;
+import net.sourceforge.ganttproject.gui.UIFacade.Choice;
+import net.sourceforge.ganttproject.resource.AssignmentContext;
+import net.sourceforge.ganttproject.resource.HumanResource;
+import net.sourceforge.ganttproject.task.ResourceAssignment;
+import net.sourceforge.ganttproject.task.ResourceAssignmentMutator;
+import net.sourceforge.ganttproject.task.Task;
+import net.sourceforge.ganttproject.util.StringUtils;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Action that adds/removes the assignment of a task to a resource
+ */
+public class AssignmentToggleAction extends GPAction {
+  private final HumanResource myHumanResource;
+
+  private final Task myTask;
+
+  public AssignmentToggleAction(HumanResource hr, Task task) {
+    super(hr.getName(), IconSize.TOOLBAR_SMALL);
+    myHumanResource = hr;
+    myTask = task;
+  }
+
+  private void createAssignment() {
+    ResourceAssignmentMutator mutator = myTask.getAssignmentCollection().createMutator();
+    ResourceAssignment newAssignment = mutator.addAssignment(myHumanResource);
+    newAssignment.setLoad(100);
+    newAssignment.setRoleForAssignment(newAssignment.getResource().getRole());
+    mutator.commit();
+  }
+
+  public void delete(HumanResource hr) {
+    ResourceAssignmentMutator mutator = myTask.getAssignmentCollection().createMutator();
+    mutator.deleteAssignment(hr);
+    mutator.commit();
+  }
+
+  @Override
+  public void actionPerformed(ActionEvent e) {
+    Boolean isSelected = (Boolean) getValue(Action.SELECTED_KEY);
```

Comment:
one more point: this should be undoable action. You can achieve it by direct using UndoManager (this will make use of UndoManager, sorry that I didn't mention it before):
```
      myUIFacade.getUndoManager().undoableEdit(getLocalizedDescription(), new Runnable() {
        @Override
        public void run() {
          try {
            // DO STUFF HERE
          } catch (Exception e) {
            getUIFacade().showErrorDialog(e);
          }
        }
      });
```

Another option is to extend class from `TaskActionBase` .

Reply:
Yes, I missed this one too. Added undo support to my patch.